### DISCUSSION TITLE
Ability to export character & timeline info to text file

### DISCUSSION THREAD

**Discussion by sshearer** [dialogic-godot/dialogic#696] [Ideas]
As a Content Creator, I would like to be able to export the definition of the characters and timelines for the purpose of review & [cross-]referencing one timeline against another for comparison/consistency purposes.

I understand that each timeline is it's own JSON file, and that there is a higher-level JSON file which identifies all the individual timelines.

However, as a Content Creator, I don't want to have to manually:
1. paste the JSON content into a text editor to remove blank lines & line breaks,
2. bold resource path and file names,
3. translate character IDs to names, and timeline IDs to names,
4. remove key-value pairs like 'event_id'.
5. add my own preceding title based on each timeline's metadata 'name' field.

I've attached an initial DRAFT sample text layout for discussion purposes.  It PURPOSELY excludes the unnecessary 'id', 'entity relationship', and 'animation-related' fields.
[Proposed Export Format.txt](https://github.com/coppolaemilio/dialogic/files/7787044/Proposed.Export.Format.txt)

A format like this will be much easier to work with in other tools than the current JSON format.

**Comment by Jowan-Spooner** [comment]
So do I understand this correctly that you only want to export this? If you find things that aren't correct, you would go back into godot and change them there? 

I'm asking because I have thought about a good way to import/export the text data of timeline into a table or text format like .csv, because that could be useful for proofreading or even translating. But that would only be useful if you could import it back into dialogic.

An export-only solution should be relatively easy. Would you want all data to go into a single file or what? How should it deal with definitions in text? Should they keep their [MyDefinition] form or be replaced with the default value of the definition?

**Reply by sshearer** [reply]
I understand the value of being able to do a 'round trip' of export/edit/import - but (as you said) for that you'll need to use something like JSON or CSV that includes entity IDs, and then how would you handle new items ?  This does definitely complicate things - and negates my desire for something 'readable' by a team.  I'd be comfortable with just editing in godot/dialogic.  For my purposes, and to avoid confusion, should we call this feature request a 'Report' instead ?  I know it's just semantics - just a thought.

I haven't used Dialogic long enough to have found a need to use definitions so I'll leave that decision to the developer(s).  If needed, I'd assume it'd appear in it's own section before the actual dialog section.  In the Dialog steps though my preference would be to just show the Definition tag and not the default value.

My preference would be for a single output file.  But maybe in the pop-up dialog where I'd define which folder to store the output there's also room to define:
a) checkboxes to define which entities to include (eg. Characters, Definitions, Dialog)
b) radio buttons for 'single file' vs. 'file per timeline' (if that is what you were thinking?)
c) I guess in theory the output filename(s) could be calculated based on the above selections.  For example:
     #projectname# - #Defn##Char##Dialog# [- #TimelineName#].txt
d) I suppose that a future evolution could include other options that could provide the support for other output formats (eg. CSV or JSON) for eventual later imports.

**Comment by sshearer** [comment]
I've done a little more work on the desired output format - cleaned it up further.  Please see the attachment:
[Proposed Export Format.txt](https://github.com/coppolaemilio/dialogic/files/7794871/Proposed.Export.Format.txt)
.

**Comment by hsandt** [comment]
How is this going on in 2024? I just found this page on localization: https://docs.dialogic.pro/translation.html that seems to hint that CSV is now supported (although not JSON, and above all gettext, see https://github.com/dialogic-godot/dialogic/issues/1874, for better versioning and localization tool support).

I'd need this to use Dialogic on any narrative project bigger than a jam, would you say that the current system of Update CSV files + Collect translations is good enough to start working on long dialogues and have them reviewed, proofread externally (I'm also thinking about automated spellchecks or even Google Doc style ML-based suggestions), translated then reinjected into Godot assets?

I may try some translation of our jam game post-jam to see how it fares.

**Reply by CakeVR** [reply]
This thread was created in 2021 for Dialogic 1. 

Dialogic 2 has been redesigned from scratch, I would recommend to ask this in a new thread or on our Discord.
There is complete CSV Translation support for Characters, Timelines, and the Glossary. The linked doc page is for Dialogic 2 only.

**Comment by CakeVR** [comment]
I will close this discussion as it has been quite some years.
